sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCacheTest.java

```diff
@@ -1284,14 +1288,18 @@ public static<T> T getSuccessResult(Mono<T> observable, long timeout) {
     }
 
     public static void validateSuccess(Mono<List<Address>> observable,
-                                       PartitionReplicasAddressesValidator validator, long timeout) {
+                                       PartitionReplicasAddressesValidator validator,
+                                       HttpClientUnderTestWrapper httpClient,
+                                       long timeout) {
         TestSubscriber<List<Address>> testSubscriber = new TestSubscriber<>();
         observable.subscribe(testSubscriber);
         testSubscriber.awaitTerminalEvent(timeout, TimeUnit.MILLISECONDS);
         testSubscriber.assertNoErrors();
         testSubscriber.assertComplete();
         testSubscriber.assertValueCount(1);
         validator.validate(testSubscriber.values().get(0));
+        // Verifying activity id is being set in header on address call to gateway.
+        assertThat(httpClient.capturedRequests.get(0).headers().value(HttpConstants.HttpHeaders.ACTIVITY_ID)).isNotNull();
```

Comment:
Can we also verify the value of the activityId header - since we are generating it.

Reply:
Done